packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -0,0 +1,297 @@
+   * @private
+   */
+  private mirrorNodeClient: MirrorNodeClient;
+
+  /**
+   * This limiter tracks hbar expenses and limits.
+   * @type {HbarLimit}
+   * @private
+   */
+  private readonly hbarLimiter: HbarLimit;
+
+  /**
+   * Histogram for capturing the cost of transactions and queries.
+   * @type {Histogram}
+   * @private
+   */
+  private readonly consensusNodeClientHistogramCost: Histogram;
+
+  /**
+   * Histogram for capturing the gas fee of transactions and queries.
+   * @type {Histogram}
+   * @private
+   */
+  private readonly consensusNodeClientHistogramGasFee: Histogram;
+
+  /**
+   * Constructs an instance of the MetricService responsible for tracking and recording various metrics
+   * related to Hedera network interactions and resource usage.
+   *
+   * @param {Logger} logger - Logger instance for logging system messages.
+   * @param {SDKClient} sdkClient - Client for interacting with the Hedera SDK.
+   * @param {MirrorNodeClient} mirrorNodeClient - Client for querying the Hedera mirror node.
+   * @param {HbarLimit} hbarLimiter - Rate limiter for managing HBAR-related operations.
+   * @param {Registry} register - Registry instance for registering metrics.
+   */
+  constructor(
+    logger: Logger,
+    sdkClient: SDKClient,
+    mirrorNodeClient: MirrorNodeClient,
+    hbarLimiter: HbarLimit,
+    register: Registry,
+  ) {
+    this.logger = logger;
+    this.sdkClient = sdkClient;
+    this.hbarLimiter = hbarLimiter;
+    this.mirrorNodeClient = mirrorNodeClient;
+    this.consensusNodeClientHistogramCost = this.initCostMetric(register);
+    this.consensusNodeClientHistogramGasFee = this.initGasMetric(register);
+  }
+
+  /**
+   * Retrieves the transaction metrics for a given transaction ID
+   * by redirecting calls to either consensus node client or mirror node client based on default configuration.
+   *
+   * @param {string} transactionId - The ID of the transaction.
+   * @param {string} callerName - The name of the entity calling the transaction.
+   * @param {string} requestId - The request ID for logging purposes.
+   * @param {string} txConstructorName - The name of the transaction constructor.
+   * @param {string} operatorAccountId - The account ID of the operator.
+   */
+  public async captureTransactionMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+    transactionType: string,
+    interactingEntity: string,
+  ) {
+    let gasUsed: number = 0;
+    let transactionFee: number = 0;
+    let txRecordChargeAmount: number = 0;
+    const formattedRequestId = formatRequestIdMessage(requestId);
+
+    // check if calls are default to consensus node or not
+    const defaultToConsensusNode = process.env.GET_RECORD_DEFAULT_TO_CONSENSUS_NODE === 'true';
+
+    // retrieve metrics
+    if (defaultToConsensusNode) {
+      const transactionRecordMetrics = await this.sdkClient.getTransactionRecordMetrics(
```

Comment:
I think you can extract the call to this method in a separate function where you check the GET_RECORD_DEFAULT_TO_CONSENSUS_NODE. Looks more readable to me like this
E.g
```
private async getTransactionRecordMetrics(
  transactionId: string,
  callerName: string,
  requestId: string,
  txConstructorName: string,
  operatorAccountId: string,
  defaultToConsensusNode: boolean,
): Promise<{ gasUsed?: number; transactionFee?: number; txRecordChargeAmount?: number } | null> {
  if (defaultToConsensusNode) {
    return this.sdkClient.getTransactionRecordMetrics(
      transactionId,
      callerName,
      requestId,
      txConstructorName,
      operatorAccountId,
    );
  } else {
    return this.mirrorNodeClient.getTransactionRecordMetrics(
      transactionId,
      callerName,
      requestId,
      txConstructorName,
      operatorAccountId,
    );
  }
}
```
Then if the result is null you can handle it and if not
you can extract the values you want
  const { gasUsed, transactionFee, txRecordChargeAmount } = transactionRecordMetrics;